src/lib.rs

```diff
@@ -446,7 +491,219 @@ extern "C" {
+    (*transfer).buffer.offset(LIBUSB_CONTROL_SETUP_SIZE as isize)
+}
+
+#[allow(non_snake_case)]
+/// Helper function to populate the setup packet (first 8 bytes of the data
+/// buffer) for a control transfer. The wIndex, wValue and wLength values should
+/// be given in host-endian byte order.
+pub unsafe fn libusb_fill_control_setup(buffer: *mut c_uchar,
+                                        bmRequestType: u8,
+                                        bRequest: u8,
+                                        wValue: u16,
+                                        wIndex: u16,
+                                        wLength: u16) {
+    let setup: &mut libusb_control_setup = std::mem::transmute(buffer);
+    setup.bmRequestType = bmRequestType;
+    setup.bRequest = bRequest;
+    setup.wValue = libusb_cpu_to_le16(wValue);
+    setup.wIndex = libusb_cpu_to_le16(wIndex);
+    setup.wLength = libusb_cpu_to_le16(wLength);
+}
+
+/// Helper function to populate the required \ref libusb_transfer fields
+/// for a control transfer.
+///
+/// If you pass a transfer buffer to this function, the first 8 bytes will
+/// be interpreted as a control setup packet, and the wLength field will be
+/// used to automatically populate the \ref libusb_transfer::length "length"
+/// field of the transfer. Therefore the recommended approach is:
+/// -# Allocate a suitably sized data buffer (including space for control setup)
+/// -# Call libusb_fill_control_setup()
+/// -# If this is a host-to-device transfer with a data stage, put the data
+///   in place after the setup packet
+/// -# Call this function
+/// -# Call libusb_submit_transfer()
+///
+/// It is also legal to pass a NULL buffer to this function, in which case this
+/// function will not attempt to populate the length field. Remember that you
+/// must then populate the buffer and length fields later.
+pub unsafe fn libusb_fill_control_transfer(transfer: *mut libusb_transfer,
+                                           dev_handle: *mut libusb_device_handle,
+                                           buffer: *mut c_uchar,
+                                           callback: libusb_transfer_cb_fn,
+                                           user_data: *mut c_void,
+                                           timeout: c_uint) {
+    // struct libusb_control_setup *setup = (struct libusb_control_setup *)(void *) buffer;
+    let setup: *mut libusb_control_setup = std::mem::transmute(buffer);
+    (*transfer).dev_handle = dev_handle;
+    (*transfer).endpoint = 0;
+    (*transfer).transfer_type = LIBUSB_TRANSFER_TYPE_CONTROL;
+    (*transfer).timeout = timeout;
+    (*transfer).buffer = buffer;
+    let ptr = setup as *const usize;
+    if !ptr.is_null()  {
+        // let size_of_result = libusb_cpu_to_le16((*setup).wLength);
+        (*transfer).length = (LIBUSB_CONTROL_SETUP_SIZE + std::mem::size_of::<u16>()).try_into().unwrap();
```

Comment:
This assumes that the length of the payload is exactly 16 bits.
For payloads of other sizes, the transfer will fail.